sdk/eventhub/Azure.Messaging.EventHubs.CheckpointStore.Blob/tests/Infrastructure/StorageScope.cs

```diff
@@ -0,0 +1,190 @@
+                // to ARM being temperamental, some management operations may be rejected.  Throwing here
+                // does not help to ensure resource cleanup only flags the test itself as a failure.
+                //
+                // If a blob container fails to be deleted, removing of the associated storage account at the end
+                // of the test run will also remove the orphan.
+            }
+            finally
+            {
+                client?.Dispose();
+            }
+
+            _disposed = true;
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to create a new blob container instance with a dynamically assigned unique name.
+        /// </summary>
+        ///
+        /// <param name="caller">The name of the calling method; this is intended to be populated by the runtime.</param>
+        ///
+        /// <returns>The <see cref="StorageScope" /> in which the test should be executed.</returns>
+        ///
+        public static async Task<StorageScope> CreateAsync([CallerMemberName] string caller = "")
+        {
+            caller = InvalidContainerCharactersExpression.Replace(caller.ToLowerInvariant(), String.Empty);
+            caller = (caller.Length < 16) ? caller : caller.Substring(0, 15);
+
+            var resourceGroup = TestEnvironment.EventHubsResourceGroup;
+            var storageAccount = StorageTestEnvironment.StorageAccountName;
+            var token = await ResourceManager.AquireManagementTokenAsync();
+
+            string CreateName() => $"{ Guid.NewGuid().ToString("D").Substring(0, 13) }-{ caller }";
+
+            using (var client = new StorageManagementClient(new TokenCredentials(token)) { SubscriptionId = TestEnvironment.EventHubsSubscription })
+            {
+                var container = await ResourceManager.CreateRetryPolicy().ExecuteAsync(() => client.BlobContainers.CreateAsync(resourceGroup, storageAccount, CreateName(), PublicAccess.None));
+                return new StorageScope(container.Name);
+            }
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to create a new Event Hubs namespace within a resource group, intended to be used as
+        ///   an ephemeral container for the Event Hub instances used in a given test run.
+        /// </summary>
+        ///
+        /// <returns>The key attributes for identifying and accessing a dynamically created Event Hubs namespace.</returns>
+        ///
+        public static async Task<StorageProperties> CreateStorageAccountAsync()
+        {
+            var subscription = TestEnvironment.EventHubsSubscription;
+            var resourceGroup = TestEnvironment.EventHubsResourceGroup;
+            var token = await ResourceManager.AquireManagementTokenAsync();
+
+            string CreateName() => $"neteventhubs{ Guid.NewGuid().ToString("N").Substring(0, 12) }";
```

Comment:
Same here. I don't think the function is necessary.

Reply:
Oh, it is.  :)      Justification above.